### ISSUE TITLE
VSCode Debugging with Arguments

### ISSUE THREAD

**Issue by spozner** [PowerShell/vscode-powershell#2995]
Please update the debugging options in VSCode to allow for either:

- keeping the last 10 unique arguments passed to the current script

- set it up so we can launch a Powershell function directly to debug execution or extend debugging.  

Ultimately, when launching a script, I'd like to be able to select the parameters available when Launching the script (Easily exposed using the Show-Command ${file} -Passthru command and arguments, or select them from a last used set of arguments for that particular file.  I've written code to do this for the PSISE and it worked great but, I'm finding the debugging interface in VSCode lacking and would like to extend it.

**Comment by TylerLeonhardt** [comment]
> set it up so we can launch a Powershell function directly to debug execution or extend debugging.

We supply a debug configuration called "PowerShell: Launch Script"
https://github.com/PowerShell/vscode-powershell/blob/ee5521fd8dd8bd1a973362facea2aa5d658d90b3/package.json#L522-L530

If you put your function... or really any script in there. That should work.

**Comment by TylerLeonhardt** [comment]
> keeping the last 10 unique arguments passed to the current script

I thought we had an issue about this but I guess not. I think it's a reasonable request.

**Comment by rkeithhill** [comment]
There is a request for this somewhere.  What I remember is that the VSCode did not support an input box (at the time) that allowed you to either select from a list OR enter arbitrary text.

**Comment by TylerLeonhardt** [comment]
@rkeithhill I feel like this might be possible now

**Comment by spozner** [comment]
@rkeithhill Were you able to find it?  Seems like an easy ask to have it remember the last x number of commands used when launching the active script...
@TylerLeonhardt Thanks for pointing me to that, I was able to use my old PSISE code to launch the active script.  I'm having a small problem setting up the hotkey for it as even F5 seems to default to that entry.  I wanted to create a new hotkey that enters debugging but uses my custom function which tracks the last 10 unique parameters used based on the script name.

The Launch.json:

"configurations": [
        {
            "name": "PowerShell: Launch Current File",
            "type": "PowerShell",
            "request": "launch",
            "script": "${file}",
            "cwd": "${file}"
        },
        {
            "name": "PowerShell Launch GetParameters Script",
            "type": "PowerShell",
            "request": "launch",
            "script": "Get-VsCodeScriptParameters",
            "cwd": "${file}"
        },
    ]

The VSCode Module I load defines the function and also loads the command to be executed in the editor.  I just cannot figure out why I can't use F5 for one and CTRL+ALT+F5 for gathering parameters.  Any insight would be helpful.

**Comment by rkeithhill** [comment]
@TylerLeonhardt What is the new API?  I was just using a new extension the other day and figured out this is what they were doing to get around the lack of this type of control:
![image](https://user-images.githubusercontent.com/5177512/97463191-1901a080-1905-11eb-9b0f-05ca6b94a273.png)
BTW this ^^ confused the heck out of me because I kept typing in a new regex and it wouldn't take it.  Turns out this is a pick list.  You either have to pick a previously entered regex, that's in the list, OR click on `New pattern...`, to get to the second level, text input box shown below:
![image](https://user-images.githubusercontent.com/5177512/97463421-58c88800-1905-11eb-8c64-b419a526dc80.png)

**Comment by TylerLeonhardt** [comment]
@rkeithhill yeah I think I've seen similar experiences and that's what I was referring to wrt an "API". That is probably still better then what we have... but I'll let @spozner chime in on that to see.

**Comment by spozner** [comment]
If that could be implemented with the debugger for executing scripts it would fill that gap nicely.
